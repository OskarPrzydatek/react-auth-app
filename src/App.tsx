import React from 'react'
import { post } from './api'
import LoginForm from './components/LoginForm'

const App: React.FC = () => {
  const [loginMessage, setLoginMessage] = React.useState<string | undefined>()

  const login = (email: string, password: string) => {
    post(`/auth?email=${email}&password=${password}`).then((data) => {
      console.log('data', data)
      setLoginMessage(data.msg)
    })
  }

  return (
    <div className="app">
      <h1 className="login-message" data-testid="login-message">
        {loginMessage ?? null}
      </h1>
      <LoginForm onLoginSubmit={login} />
    </div>
  )
}

export default App
