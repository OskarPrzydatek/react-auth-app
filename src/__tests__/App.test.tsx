import { describe, test } from 'vitest'
import { act, fireEvent, render, screen } from '@testing-library/react'
import App from '../App'

describe('App', () => {
  test('Snapshot ', () => {
    const view = render(<App />)
    expect(view).toMatchSnapshot()
  })

  test('ensure login form validation works correctly', async () => {
    render(<App />)

    // Empty form validation
    act(() => {
      fireEvent.click(screen.getByTestId('login-form-submit-button'))
    })
    expect(
      await screen.findByTestId('login-form-email-validation-message'),
    ).toHaveTextContent('Email is required')
    expect(
      await screen.findByTestId('login-form-password-validation-message'),
    ).toHaveTextContent('Password is required')

    // Email format validation
    act(() => {
      fireEvent.change(screen.getByTestId('login-form-email-text-field'), {
        target: { value: 'oskar.m.przydatek' },
      })
      fireEvent.change(screen.getByTestId('login-form-password-text-field'), {
        target: { value: 'Strong!123' },
      })
    })
    expect(
      await screen.findByTestId('login-form-email-validation-message'),
    ).toHaveTextContent('Please put email in valid form')

    // Correct filed form
    act(() => {
      fireEvent.change(screen.getByTestId('login-form-email-text-field'), {
        target: { value: 'oskar.m.przydatek@gmail.com' },
      })
      fireEvent.click(screen.getByTestId('login-form-submit-button'))
    })
    expect(
      await screen.findByTestId('login-form-email-validation-message'),
    ).not.toBeInTheDocument()
  })
})
