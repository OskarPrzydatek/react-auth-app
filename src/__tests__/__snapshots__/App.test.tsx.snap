// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Snapshot  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <h1
          class="login-message"
          data-testid="login-message"
        />
        <form
          class="form"
        >
          <h2
            class="form__header"
          >
            Zaloguj Się
          </h2>
          <label
            class="text-field"
          >
            <input
              class="input-clean text-field__input"
              data-testid="login-form-email-text-field"
              name="email"
              placeholder="email"
            />
          </label>
          <label
            class="text-field"
          >
            <input
              class="input-clean text-field__input"
              data-testid="login-form-password-text-field"
              name="password"
              placeholder="password"
              type="password"
            />
          </label>
          <input
            class="input-clean submit-button"
            data-testid="login-form-submit-button"
            type="submit"
            value="Login"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <h1
        class="login-message"
        data-testid="login-message"
      />
      <form
        class="form"
      >
        <h2
          class="form__header"
        >
          Zaloguj Się
        </h2>
        <label
          class="text-field"
        >
          <input
            class="input-clean text-field__input"
            data-testid="login-form-email-text-field"
            name="email"
            placeholder="email"
          />
        </label>
        <label
          class="text-field"
        >
          <input
            class="input-clean text-field__input"
            data-testid="login-form-password-text-field"
            name="password"
            placeholder="password"
            type="password"
          />
        </label>
        <input
          class="input-clean submit-button"
          data-testid="login-form-submit-button"
          type="submit"
          value="Login"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
